src/Apps/Artwork/Components/ArtworkDetails/RequestConditionReport.tsx

```diff
@@ -0,0 +1,205 @@
+import { Button, Col, Modal, Row, Sans, Serif, Spinner } from "@artsy/palette"
+import React, { useState } from "react"
+import { commitMutation, graphql, RelayProp } from "react-relay"
+
+import { AnalyticsSchema, track, useSystemContext, useTracking } from "Artsy"
+import * as Schema from "Artsy/Analytics/Schema"
+import { ErrorModal } from "Components/Modal/ErrorModal"
+import * as authentication from "Components/NavBar/Utils/authentication"
+import createLogger from "Utils/logger"
+
+import { RequestConditionReportMutation } from "__generated__/RequestConditionReportMutation.graphql"
+
+const logger = createLogger(
+  "Apps/Artwork/Components/ArtworkDetails/RequestConditionReport"
+)
+
+export interface RequestConditionReportProps {
+  saleArtworkID: string
+  relay?: RelayProp
+}
+
+export const RequestConditionReport: React.FC<RequestConditionReportProps> = track(
+  {
+    context_module: AnalyticsSchema.ContextModule.RequestConditionReport,
+  }
+)(props => {
+  const { mediator, user } = useSystemContext()
+  const { trackEvent } = useTracking()
+
+  const [requestingConditionReport, setRequestingConditionReport] = useState(
+    false
+  )
+
+  const [
+    showConditionReportRequestedModal,
+    setShowConditionReportRequestedModal,
+  ] = useState(false)
+
+  const [showErrorModal, setShowErrorModal] = useState(false)
+
+  const isLoggedIn = Boolean(user)
+  const { saleArtworkID, relay } = props
+
+  const requestConditionReport = () => {
+    return new Promise(async (resolve, reject) => {
```

Comment:
Have you considered typing the promise returned here, similar to what we do in the [`ConfirmBid` component?](https://github.com/artsy/reaction/blob/f53215c82577ef229b6dec192211d8a9175b5f14/src/Apps/Auction/Routes/ConfirmBid/index.tsx#L79). I _believe_ that would remove needing the cast to `any` on L122 below.

Reply:
Definitely thinking about tightening up the types. Haven't gotten to it yet but will definitely take a look at this promise while I'm doing so.

Reply:
Added that promise type :+1: